### ISSUE TITLE
TEDLIUM silence padding of segments

### ISSUE THREAD

**Issue by danpovey** [kaldi-asr/kaldi#835]
I just had a look at the alignments in the TEDLIUM setup, and I noticed that there seems to be no silence at the beginning and end of most segments.  I think this is a bad thing, and it might be the reason why the chain model is not working well there.
I am going to write a script to analyze the alignments that can be called at the end of the GMM training passes.
What should be done is just to pad the beginning and end of segments with a little silence e.g. 0.1 seconds, and then afterwards fix them up to make sure there are no overlaps between segments and none of them are negative.  @galv can you do this?

**Comment by galv** [comment]
I'm not sure if I can do it. Does #777 handle adding silence padding as you describe?

My problem is that if it involves a lot of coding, I may not be able to do this in a timely fashion right now. There are a lot of decisions, and which one is optimal is not clear here. For example, would the padded silence at the beginning and ending be 0.1 seconds of simply "0" as the magnitude in the wave time series? Or would the silence audio be background noise somehow obtained by speaker diarization?

Alternatively, are you talking about adding "silence" as in a silence label to the beginning and end of every utterance transcription? I think this may be what you mean when you say "segment."

**Comment by danpovey** [comment]
All I meant was adding/subtracting a little from the end/begin times in the
segments in the segments file.  It's a fairly small change to the data
preparation script.

**Comment by galv** [comment]
That is simple enough, sounds fine.

**Comment by galv** [comment]
That is to say, I will work on it.

**Comment by danpovey** [comment]
BTW, it may be necessary to do the same for the test set.
I know it seems a bit odd to mess with the test-set segmentation, but if
they have provided a segmentation with no silence padding, it may not be
totally suitable for normal processing.

Incidentally, the just-now-added scripts
`steps/diagnostic/analyze_alignments.sh` and
`steps/diagnostic/analyze_lats.sh` will be helpful for looking at the
amount of silences present in the training and test sets respectively.
They are now called from training and decoding scripts, but once you
update, you can them manually on previously-created directories.

Dan

**Comment by galv** [comment]
@vimalmanohar Can you confirm whether or not your data clean up scripts do something similar to this silence padding? I want to know whether I should prioritize this to finish in time for our camera ready paper.

**Comment by danpovey** [comment]
They won't do this- the data cleanup scripts produce a sub-segmentation of
an existing segmenation, so they won't add silence that wasn't there
already.
BTW, it would be good if you could run an alignment of the training data,
and maybe a decode, and share some of the stats that those scripts I
recently added, in steps/diagnostic/, will produce.  Or run those scripts
directly to get the stats from an existing alignment.

**Comment by galv** [comment]
@danpovey I see you self-assigned this. Are you working on this now? I was about to get back to this. Of course, I don't mind being relieved of it ;D

**Comment by danpovey** [comment]
Yes.
I did it already-- but I'll check it in together with some other stuff I'm
working on regarding Vimal's data cleanup.
I'm working in /home/dpovey/kaldi-trunk/egs/tedlium/s5.
Padding the training segments was helpful-- it was about 0.3% to 1%
depending on the test set and the model (only tested with the GMM models).
Dan
